### DISCUSSION TITLE
Missing element description for [model]

### DISCUSSION THREAD

**Discussion by Jason-Zhou1** [osrf/rmf_core#235] [Q&A]
Hi all,
I am using Ubuntu20.04 and gazebo 11 to simulate the office delivery task.When i launch the delivery file, i get error messages:

[gzclient-11] [Wrn] [Event.cc:61] Warning: Deleting a connection right after creation. Make sure to save the ConnectionPtr from a Connect call
[gzclient-11] [INFO] [1608198689.212026115] [gazebo_ros_node]: ROS was initialized without arguments.
[gzclient-11] ToggleCharging::ToggleCharging()
[gzclient-11] ToggleFloors::ToggleFloors()
[gzclient-11] ToggleFloors::Load()
[gzclient-11] Error [Element.cc:796] Missing element description for [model]

The rviz display is normal, but when i launch loop.launch.xml file, the tinyrobots do not move anyway.
May I know how to solve this issue?

**Comment by arjo129** [comment]
Hi Jason,
There could be a few reasons for this.
Can you see the robots in gazebo?
There could be multiple reasons for this. First of all in the rviz panel are there any green lines that get drawn? 
Can you manually use the `rmf_panel` to send commands to the robots?
Also when did you last pull from the git repos?

**Comment by Jason-Zhou1** [comment]
Hi Arjo,
The robot is not seen in gazebo. In rviz the green lines appear as the planned path.
I manually use rmf_panel to send commands to robot, but the results is same. The messages in command line are as following:
```
[rviz2-7] [INFO] [1608259157.786648232] [rmf_panel]: Queued loop request
[rviz2-7] 10:39:17	Loop	Fleet: tinyRobot	From: wp24	To: wp30Repeat: 2	
[rviz2-7] 
[full_control-12] Queuing new task for [tinyRobot/tinyRobot2]. New queue size: 1
[full_control-12] [INFO] [1608259158.581269362] [tinyRobot_fleet_adapter]: Queuing new task [6d215c2f] for [tinyRobot/tinyRobot2]. New queue size: 1
[full_control-12] [INFO] [1608259158.581302403] [tinyRobot_fleet_adapter]: Beginning new task [6d215c2f] for [tinyRobot/tinyRobot2]. Remaining queue size: 0
[gzserver-10] [INFO] [1608259158.582988399] [slotcar_tinyRobot2]: tinyRobot2 received a path request with 5 waypoints
[gzserver-10] [INFO] [1608259159.006142905] [slotcar_tinyRobot2]: 1 already received task [tinyRobot2] -- continuing as normal
[gzserver-10] [INFO] [1608259159.225714054] [slotcar_tinyRobot2]: tinyRobot2 reached waypoint 1/4
[full_control-12] [INFO] [1608259173.053081563] [tinyRobot_fleet_adapter]: Replanning for [tinyRobot/tinyRobot2] because of an interruption
[gzserver-10] [INFO] [1608259173.054086943] [slotcar_tinyRobot2]: tinyRobot2 received a path request with 4 waypoints
```
This procedure repeats all the way.

**Comment by youliangtan** [comment]
If the robot doesn't appear in gazebo, this should be related to the sourcing of the `tinyRobot` model path.  Try to recompile 
`rmf_demo_assets` with the following command `colcon build --packages-select rmf_demo_assets`.

Other than that, this issue is closely related to [rmf_demos](https://github.com/osrf/rmf_demos/). Kindly request all future issues can be raised at the relevant repo. Thanks

**Comment by Jason-Zhou1** [comment]
Hi tanyouliang,
I have run the command:colcon build --packages-select rmf_demo_assets
It passes the build with no stderr.

**Comment by arjo129** [comment]
Have you updated your local repository to the latest commit? In particulat you may want to update `traffic_editor` so that the slotcar plugins get updated as well.

**Comment by Jason-Zhou1** [comment]
Hi Arjo,
I tried to reinstall the rmf_demos_ws. When i build it by following command: 
$ cd ~/rmf_demos_ws
$ rosdep install --from-paths src --ignore-src --rosdistro foxy -yr

I get error message:
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
rmf_traffic: Cannot locate rosdep definition for [fcl]
Continuing to install resolvable dependencies...
#All required rosdeps installed successfully

**Comment by arjo129** [comment]
Hi Jason, `rosdep` is for pulling binary dependencies. You should either use `git pull` or `vcs pull` to update the code in the repos. Could you perhaps share your build setup?
